napari/conftest.py

```diff
@@ -434,3 +437,41 @@ def disable_notification_dismiss_timer(monkeypatch):
         monkeypatch.setattr(NapariQtNotification, "DISMISS_AFTER", 0)
         monkeypatch.setattr(NapariQtNotification, "FADE_IN_RATE", 0)
         monkeypatch.setattr(NapariQtNotification, "FADE_OUT_RATE", 0)
+
+
+@pytest.fixture
+def linux_wm():
+    """Start a WM in the background for tests that need a WM.
+
+    This is required for tests that depend on UI events (e.g. setFocus).
+    See https://github.com/pytest-dev/pytest-qt/issues/206.
+
+    This will only run on Linux and in CI (determined by CI env var) - this is
+    unnecessary/irrelevant on macOS and Windows.
+
+    The window manager start command can be set via env var `WM_START_CMD`
+    (default is `herbstluftwm` - make sure it's installed).
+    """
+    wm_start_cmd = os.environ.get("WM_START_CMD", "herbstluftwm")
+    proc = None
+    if platform.system() == "Linux" and os.environ.get("CI"):
+        proc = subprocess.Popen([wm_start_cmd])
+        # give the WM 1s to start up and check it's still running
+        time.sleep(1)
+        if proc.poll() is not None:
+            raise RuntimeError(
+                f"window manager '{wm_start_cmd}' process [{proc.pid}] exited, "
+                f"return code: {proc.returncode}"
+            )
+
+    yield proc
+
+    if proc:
+        proc.terminate()
+        try:
+            proc.wait(timeout=20)
+        except subprocess.TimeoutExpired:
+            proc.kill()
+            raise RuntimeError(
+                f"failed to terminate window manager '{wm_start_cmd}'"
+            )
```

Comment:
This fixture is effectively duplicate code with `napari-console` - maybe it should be moved to its own pytest plugin? Would napari plugin developers benefit from something like `pytest-napari` for example?

Reply:
as napari-console depends on napari then you could move this test to `napari/utils/_testsupport.py`

https://github.com/napari/napari/blob/main/napari/utils/_testsupport.py

as napari declares pytest plugins in this file there will be no need to create a separate plugin

https://github.com/napari/napari/blob/51253c09d827393886ba7336fedc2efb82e1ee1d/setup.cfg#L165

Unfortunately, until napari 0.5.0 (the next release) the copy in `napari-console` will be required, but I do not see this as a problem.

Reply:
Oh nice - thanks! I will make a note to remove the version in `napari-console` on the next release.

I still may look into adding it to the GitHub Action instead if I find some time. I don't really like that this basically introduces a hidden system dependency, even if only in CI.